quark/core/shurikenapkinfo.py

```diff
@@ -0,0 +1,532 @@
+        methodAnalysis = method_object.cache
+
+        lowerFuncs = []
+        for i in range(methodAnalysis.n_of_xrefto):
+            xref = methodAnalysis.xrefto[i]
+            lowerFuncs.append(
+                (
+                    self._convert_to_method_object(xref.method.contents),
+                    xref.idx,
+                )
+            )
+
+        return lowerFuncs
+
+    def get_method_bytecode(
+        self, method_object: MethodObject
+    ) -> Iterator[BytecodeObject]:
+        """
+        Inherited from baseapkinfo.py.
+        Return the bytecodes of the specified method.
+
+        :param method_object: a target method to get the corresponding
+        bytecodes
+        :yield: a generator of BytecodeObjects
+        """
+        disassembledMethod = self.__getDisassembledMethod(method_object)
+
+        for i in range(disassembledMethod.n_of_instructions):
+            rawBytecode = disassembledMethod.instructions[
+                i
+            ].disassembly.decode(errors="backslashreplace")
+            yield self._parseSmali(rawBytecode)
+
+    def _parseParameters(self, parameter: str) -> Union[int, float, str]:
+
+        if parameter[:2] == "0x":
+            try:
+                parameter = int(parameter, 16)
+                return parameter
+            except (TypeError, ValueError):
+                pass
+
+        try:
+            parameter = int(parameter, 10)
+            return parameter
+        except (TypeError, ValueError):
+            pass
+        try:
+            parameter = float(parameter)
+            return parameter
+        except (TypeError, ValueError):
+            pass
+
+        typeTable = {
+            "void": "V",
+            "boolean": "Z",
+            "byte": "B",
+            "short": "S",
+            "char": "C",
+            "int": "I",
+            "long": "J",
+            "float": "F",
+            "double": "D",
+        }
+        for typeName, abbreviation in typeTable.items():
+            parameter = parameter.strip()
+            pattern = r" ({})(\[\])*$".format(typeName)
+            if re.search(pattern, parameter):
+                parameter = re.sub(
+                    pattern, r" {}\2".format(abbreviation), parameter
+                )
+                break
+
+        patternToIdentifyMemberField = r"->\w+(?!\(\)) "
+        if re.search(patternToIdentifyMemberField, parameter):
+            parameter = self._convertMemberFieldFormat(parameter)
+
+        return parameter
+
+    def _parseSmali(self, smali: str) -> BytecodeObject:
+
+        smali = smali.split("//")[0].strip()
+        if smali == "":
+            raise ValueError("Argument cannot be empty.")
+
+        if " " not in smali:
+            return BytecodeObject(smali, None, None)
+
+        mnemonic, args = smali.split(maxsplit=1)
+        parameter = None
+
+        # extract string
+        quoteChar = args[-1]
+        if quoteChar == '"' or quoteChar == "'":
+
```

Comment:
I recommend checking `args[-1]` before declaring `quoteChar` because we don't need `quoteChar` if `smali` has no string.

```suggestion
        if args[-1] == '"' or args[-1] == "'":
            # Extract string
            quoteChar = args[-1]

```